### PR TITLE
Assignments 1 and 2

### PR DESCRIPTION
**Pull request by Sir-NoChill**
Modified resources for assignments 1 and 2
- Switched SPEC benchmarks to RISCV
- Switched GEM5 compilation instructions
- Added required templates

**NOTE**: I will have a lot of explaining to do in the videos. My explanations will cover how to use these materials in the student assignments

### CODE REVIEW

File: scripts/SiFiveP550/SiFiveP550.py [hunk 1]
```diff
@@ -86,14 +89,16 @@
 SimpleOpts.add_option("--binary", nargs="?", default=default_binary)
 
 # Arguments to the binary
-SimpleOpts.add_option("--input", default="")
-
-# Any additional flags
-SimpleOpts.add_option("--flags", default="")
+SimpleOpts.add_option("--input", default="", help="Use this if you need a file")
```
Comment (madisonbradley112) [inline scripts/SiFiveP550/SiFiveP550.py]: nitpick/wording: could make the help text more precise, eg. 
- `"Specify binary file to execute on the simulated processor"`
- `"Specify input file for the executed binary on the simulated processor"`
- `"Specify arguments for the executed binary on the simulated processor"`
- `"Specify SPEC CPU 2017 benchmark to run (NOTE: will override --input and --args with SPEC inputs and args)"`

File: scripts/SiFiveP550/SiFiveP550.py [hunk 2]
```diff
@@ -203,10 +220,12 @@
 if args.maxinsts:
     for i in range(args.nprocs):
         system.cpu[i].max_insts_any_thread = args.maxinsts
+else:
+    args.maxinsts = "all of the"
```
Comment (madisonbradley112) [inline scripts/SiFiveP550/SiFiveP550.py]: nitpick: I was a bit confused following the logic at first glance w/ the "all of the" string: i think maxinsts should remain as a conceptually numerical value until it's printed 

Could move this if-statement down to the printing at the bottom instead

File: scripts/Spec.py
Comment (madisonbradley112) [inline scripts/Spec.py:1]: issue: this file is duplicated under both scripts/SiFiveP550/Spec.py and here

File: scripts/tutorial/02_caches/caches.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    tgts_per_mshr = 20
+
+    def connectCPU(self, cpu):
+        raise NotImplementedError
+
+    def connectBus(self, bus):
+        self.mem_side = bus.cpu_side_ports
+
+
+# Vary the L1I cache and L1Dcaceh based on that
```
Comment (madisonbradley112) [inline scripts/tutorial/02_caches/caches.py]: typo: L1Dcaceh -> L1Dcache

File: scripts/Spec.py
Comment (Sir-NoChill) [reply scripts/Spec.py:1]: now a symlink

**Comment by Sir-NoChill** [comment]
Alright, I think this one is good to go if you want to give a final look over

**Comment by madisonbradley112** [comment]
would it be possible to change the submodule version here to 24.0.0.0

**Comment by Sir-NoChill** [comment]
fixed in main

**Comment by madisonbradley112** [comment]
sweet LGTM

**Pull request merged**